src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -410,7 +410,26 @@ public static string GetGeometryFactoryPath2(string rootFolder, Version version)
 
             return assemblyPath;
         }
-
+        /// <summary>
+        /// This method will return the path to the GeometryFactory assembly location for a requested version of the geometry library.
+        /// This method is tolerant to the requested version in that it will attempt to locate an exact or lower version of the GeometryFactory assembly.
+        /// </summary>
+        /// <param name="rootFolder">Full path of the directory that contains 
+        /// LibG_xxx_y_z folder, where 'xxx y z' represents the library version of asm. In a 
+        /// typical setup this would be the same directory that contains Dynamo 
+        /// core modules. This must represent a valid directory - it cannot be null.</param>
+        /// <param name="tolerantVersion">Version number of the targeted geometry library.
+        /// If the resulting assembly does not exist, this method will look for a lower version match.
+        /// This parameter cannot be null. </param>
+        /// <returns>The full path to GeometryFactoryAssembly assembly.</returns>
+        /// 
+        public static string GetGeometryFactoryPathTolerant(string rootFolder, Version tolerantVersion){
```

Comment:
Should not `tolerantVersion` parameter be more appropriately named `targetVersion`? The API caller passes in a targetted version that he expects to target.